x-pack/docs/en/rest-api/security/get-tokens.asciidoc

```diff
@@ -0,0 +1,126 @@
+[role="xpack"]
+[[security-api-get-token]]
+=== Get tokens API
+
+Creates bearer tokens for access without requiring basic authentication.
+
+==== Request
+
+`POST /_xpack/security/oauth2/token` 
+
+==== Description
+
+The tokens are created by the {es} Token Service, which is automatically enabled
+when you configure TLS on the HTTP interface. See <<tls-http>>. Alternatively,
+you can explicitly enable the `xpack.security.authc.token.enabled` setting. When 
+you are running in production mode, a bootstrap check prevents you from enabling 
+the token service unless you also enable TLS on the HTTP interface. 
+
+The get tokens API takes the same parameters as a typical OAuth 2.0 token API
+except for the use of a JSON request body.
+
+A successful get tokens API call returns a JSON structure that contains the access
+token, the amount of time (seconds) that the token expires in, the type, and the
+scope if available.
+
+The tokens returned by the get tokens API have a finite period of time for which
+they are valid and after that time period, they can no longer be used. That time
+period is defined by the `xpack.security.authc.token.timeout` setting. For more
+information, see <<token-service-settings>>.
+
+If you want to invalidate a token immediately, you can do so by using the 
+<<security-api-invalidate-token,delete tokens API>>.
+
+
+==== Request Body
+
+The following parameters can be specified in the body of a POST request and
+pertain to creating a token:
+
+`grant_type`::
+(string) The type of grant. Currently only the `password` grant type is supported.
+
+//TBD: What about the `refresh_token` type in the example below?
+
+`password` (required)::
+(string) The user's password.
+
+//TBD: Add `refresh_token` parameter per example below?
```

Comment:
Yes we need to add this as well. When grant type is refresh_token then username and password are not required but the refresh_token field is required.